### PR TITLE
deprecate Taxa names -> Features

### PR DESCRIPTION
**Pull request by himmil**

### CODE REVIEW

**Comment by himmil** [comment]
@TuomasBorman can you merge?

**Comment by TuomasBorman** [comment]
Tests (tests/testthat directory) seems still have deprecated functions

**Comment by TuomasBorman** [comment]
To replace `aes_string`:

There are two cases where we have used `aes_string`: "the standard way", i.e. case 1, and  with `do.call` function, i.e., case 3

Below you will find, how to replace these

```
library(ggplot2)

data(cars)


######################### CASE 1
p <- ggplot(cars)

p + geom_point(aes(x = speed, y = dist))

p + geom_point(aes_string(x = "speed", y = "dist"))

p + geom_point(aes(x = .data[["speed"]], y = .data[["dist"]]))

######################### CASE2

p + aes(x = speed, y = dist) + geom_point()

p + aes_string(x = "speed", y = "dist") + geom_point()

p + aes(x = .data[["speed"]], y = .data[["dist"]]) + geom_point()

############################ CASE 2.1

aesthetic <- aes(x = speed, y = dist)
p + aesthetic + geom_point()

aesthetic <- aes_string(x = "speed", y = "dist")
p + aesthetic + geom_point()


aesthetic <- aes(x = .data[["speed"]], y = .data[["dist"]])
p + aesthetic + geom_point()

################# CASE 3

# IS not working
args <- list(x = "speed", y = "dist")
aesthetic <- do.call(aes, args)
p + aesthetic + geom_point()

# Works
args <- list(x = "speed", y = "dist")
aesthetic <- do.call(aes_string, args)
p + aesthetic + geom_point()

# Option 1 (I think this is better)
args <- list(x = "speed", y = "dist")
args <- lapply(args, function(x) if (!is.null(x)) sym(x))
aesthetic <- do.call(aes, args)
p + aesthetic + geom_point()

# Option 2
args <- list(x = "speed", y = "dist")
args <- lapply(args, function(x) if (!is.null(x)) sym(x))
aesthetic <- aes(!!!args)
p + aesthetic + geom_point()

```

File: R/plotColTile.R [hunk 1]
```diff
@@ -99,19 +99,19 @@ setMethod("plotRowTile", signature = c("SummarizedExperiment"),
...
     x_group <- data %>% 
-        group_by(.data$X) %>% 
+        group_by(X) %>% 
         summarise(group_n = n()) %>%
-        mutate(group_freq = .data$group_n/sum(.data$group_n),
-               x = cumsum(.data$group_freq),
-               xmin = c(0,.data$x[-length(.data$x)]))
+        mutate(group_freq = group_n/sum(group_n),
+               x = cumsum(group_freq),
+               xmin = c(0,x[-length(x)]))
```
Comment (TuomasBorman) [inline R/plotColTile.R:106]: This should still work, so need to change?

File: R/plotColTile.R [hunk 2]
```diff
@@ -99,19 +99,19 @@ setMethod("plotRowTile", signature = c("SummarizedExperiment"),
...
         summarise(fill_n = n(), .groups = "rowwise") %>%
         dplyr::left_join(x_group, by = "X") %>%
-        mutate(fill_freq = .data$fill_n/.data$group_n) %>%
-        group_by(.data$X) %>%
-        mutate(y = cumsum(.data$fill_freq),
-               ymin = c(0,.data$y[-length(.data$y)])) %>%
+        mutate(fill_freq = fill_n/group_n) %>%
+        group_by(X) %>%
+        mutate(y = cumsum(fill_freq),
+               ymin = c(0,y[-length(y)])) %>%
```
Comment (TuomasBorman) [inline R/plotColTile.R:114]: Same here?

File: R/plotColTile.R [hunk 3]
```diff
@@ -139,9 +139,9 @@ setMethod("plotRowTile", signature = c("SummarizedExperiment"),
 .get_xcoord_mid <- function(data){
     data %>% 
         ungroup() %>%
-        select(.data$X, .data$x, .data$xmin) %>% 
+        select(X, x, xmin) %>% 
         unique() %>%
-        mutate(xmid = .data$xmin + (.data$x - .data$xmin)/2 )
+        mutate(xmid = xmin + (x - xmin)/2 )
 }
```
Comment (TuomasBorman) [inline R/plotColTile.R:146]: And here??

File: R/plotPrevalence.R [hunk 4]
```diff
@@ -485,7 +486,7 @@ setMethod("plotTaxaPrevalence", signature = c(x = "SummarizedExperiment"),
                                     size_by = NULL,
                                     alpha = line_alpha,
                                     linetype = line_type,
-                                    size = line_size)
+                                    linewidth = line_size)
```
Comment (TuomasBorman) [inline R/plotPrevalence.R:489]: Why the name has changed?

File: R/plotSeries.R [hunk 5]
```diff
@@ -313,13 +313,13 @@ setMethod("plotSeries", signature = c(object = "SummarizedExperiment"),
...
     }
-    # Fetches arguments fpr geom_line
+    # Fetches arguments for geom_line
     line_args <- .get_line_args(colour_by = colour_by,
                                 linetype_by = linetype_by,
                                 size_by = size_by,
                                 alpha = line_alpha,
                                 linetype = line_type,
-                                size = line_width)
+                                linewidth = line_width)
```
Comment (TuomasBorman) [inline R/plotSeries.R:322]: Wjy the name has changed? Was there a bug?

File: R/plotSeries.R [hunk 6]
```diff
@@ -359,7 +359,7 @@ setMethod("plotSeries", signature = c(object = "SummarizedExperiment"),
         guide_args$linetype <- guide_legend(title = linetype_by)
     }
     if (!is.null(size_by)) {
-        guide_args$size <- guide_legend(title = size_by)
+        guide_args$linewidth <- guide_legend(title = size_by)
```
Comment (TuomasBorman) [inline R/plotSeries.R:362]: This also

File: R/treeData.R [hunk 7]
```diff
@@ -118,7 +118,7 @@ setMethod("rowTreeData", signature = c(x = "TreeSummarizedExperiment"),
 DEFAULT_TREE_DATA_COLS <- c("parent","node","branch.length","label")
 .clean_tree_data <- function(tree_data){
     tree_data %>%
-        select(DEFAULT_TREE_DATA_COLS)
+        select(all_of(DEFAULT_TREE_DATA_COLS))
```
Comment (TuomasBorman) [inline R/treeData.R:121]: Was there a bug or why this has changed?

File: R/utils_plotting.R [hunk 8]
```diff
@@ -398,23 +397,23 @@ NULL
...
+    aes_args <- aes()
     if (!is.null(colour_by)) {
-        aes_args$colour <- "colour_by"
-        aes_args$fill <- "colour_by"
+        aes_args$colour <- substitute(`colour_by`)
+        aes_args$fill <- substitute(`colour_by`)
     }
-    new_aes <- do.call(aes_string, aes_args)
+    new_aes <- do.call(aes, aes_args)
     geom_args <- list(mapping = new_aes,
```
Comment (TuomasBorman) [inline R/utils_plotting.R:417]: To get this working with as little modifications as possible (which is desirable), you can just add there

args <- lapply(args, function(x) if (!is.null(x)) sym(x))

Before do.call

File: R/plotColTile.R (hunk 1 above)
Comment (himmil) [reply R/plotColTile.R:106]: .data in tidyselect expressions was deprecated in 1.2.0

File: R/plotPrevalence.R (hunk 4 above)
Comment (himmil) [reply R/plotPrevalence.R:489]: size should be replaced with linewidth in ggplot2 3.4.0

File: R/treeData.R (hunk 7 above)
Comment (himmil) [reply R/treeData.R:121]: Not a bug, but tidyselect 1.2.0 has started to require it

**Comment by TuomasBorman** [comment]
Error occurs when the data is loaded

 no visible global function --> add importFrom tags (check other functions for example)

File: R/utils_plotting.R [hunk 9]
```diff
@@ -402,19 +405,21 @@ NULL
...
+    aes_args <- lapply(aes_args, function(x) if (!is.null(x)) sym(x))
+    new_aes <- do.call(aes, aes_args)
     geom_args <- list(mapping = new_aes, alpha = alpha, colour = colour)
     return(list(args = geom_args))
 }
 
 .get_density_args <- function(colour_by, alpha = 0.65, colour = "black") {
     fill_colour <- TRUE
-    aes_args <- list()
+    aes_args <- aes()
```
Comment (TuomasBorman) [inline R/utils_plotting.R:416]: This line should still be list()

**Comment by himmil** [comment]
Now, finally, it should be ready to be merged.

**Comment by TuomasBorman** [comment]
Sorry for late reply. Can you run BiocCheck::BiocCheck(). It seems that the indentation is off (should be multiplication of 4 spaces)

**Comment by himmil** [comment]
Hi, I ran the check and it gave an error that installation calls found in vignettes. How should I fix it?

**Comment by TuomasBorman** [comment]
You can skip those errors not directly related to this PR. We can fix those things in other PRs

**Comment by himmil** [comment]
Ok, this is ready to be merged (I don't have write access to do it).

**Pull request merged**